validator/src/validator.rs

```diff
@@ -816,11 +854,12 @@ where
         // Process blockchain updates.
         let mut received_event: Option<Blake2bHash> = None;
         while let Poll::Ready(Some(event)) = self.blockchain_event_rx.poll_next_unpin(cx) {
-            let consensus_established = self.consensus.is_established();
-            trace!(?event, consensus_established, "blockchain event");
-            if consensus_established {
-                let latest_hash = event.get_newest_hash();
-                self.on_blockchain_event(event);
+            self.update_can_enforce_validity_window();
+            let can_be_active = self.can_be_active();
+            trace!(?event, can_be_active, "blockchain event");
+            let latest_hash = event.get_newest_hash();
+            self.on_blockchain_event(event);
```

Comment:
It seems fine to me, but just to make sure:
This change executes `on_blockchain_event(event)` for every event, even when `!can_be_active`. Previously it would only execute with `consensus_established`. Is that by design?